asmcomp/asmgen.ml

```diff
@@ -125,10 +125,19 @@ let compile_unit ~source_provenance asm_filename keep_asm obj_filename gen =
     raise exn
 
 let gen_implementation ?toplevel ~source_provenance ppf (size, lam) =
+  let main_module_block =
+    Clambda.{
```

Comment:
Also, I think the older syntax is a more precise control of scope, since the module qualification only applies to the label.  Pierre, please change this as @alainfrisch suggests.